core/tests/puppeteer-acceptance-tests/user-utilities/exploration-editor-utils.ts

```diff
@@ -650,8 +672,9 @@ export class ExplorationEditor extends BaseUser {
     });
     await Promise.all([
       this.clickOn(confirmDiscardButton),
-      this.page.waitForNavigation(),
+      this.page.waitForNavigation({waitUntil: 'networkidle0'}),
     ]);
+    await this.page.waitForFunction('document.readyState === "complete"');
```

Comment:
Can you abstract this to a utility function like waitForPageToFullyLoad() or something like that?

Reply:
Done